### PR TITLE
[Maybank2u.com.ph] New ruleset

### PR DESCRIPTION
**Pull request by youdly**
https://github.com/EFForg/https-everywhere/pull/14066

### CODE REVIEW

**Comment by youdly** [comment]
Rebased to resolve build issue

**Comment by youdly** [comment]
Added test URLs and rebased.

**Comment by youdly** [comment]
travis error in chrome beta

**Comment by J0WI** [comment]
This will still fail if you try to navigate to the site, because of the 404 on https (e.g. on `https://www.maybank2u.com.ph`).

**Comment by youdly** [comment]
Excluded `www.maybank2u.com.ph/$`

**Comment by J0WI** [comment]
I tried it on some random pages, but I'm always getting 404s on https. Can you find any regex for the rule that works? We should avoid the generic rule here.

**Comment by youdly** [comment]
Excluded all directories except functional's.

File: src/chrome/content/rules/Maybank2u.com.ph.xml [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
 		<test url="http://maybank2u.com.ph/resources/my/img/common/logo-maybank_m.png" />
 		<test url="http://www.maybank2u.com.ph/resources/my/img/common/logo-maybank2u_m.png" />
 
-	<!-- 404 on main page -->
-	<exclusion pattern="^http://(www\.)?maybank2u\.com\.ph/$" />
-
 	<securecookie host=".+" name=".+" />
 
+	<!-- only RIBNOW, resources and ph_maybank_gif directories are functional -->
+	<exclusion pattern="^http://(www\.)?maybank2u\.com\.ph/(?!RIBNOW|resources|ph_maybank_gif)" />
```
Comment (J0WI) [inline src/chrome/content/rules/Maybank2u.com.ph.xml]: The negative lookahead operator should be [avoided](https://github.com/EFForg/https-everywhere/blob/master/CONTRIBUTING.md#regex-conventions) in favor of a more restrictive redirect rule.

Comment (J0WI) [reply src/chrome/content/rules/Maybank2u.com.ph.xml]: And a trailing slash is missing.

File: src/chrome/content/rules/Maybank2u.com.ph.xml [hunk 2]
```diff
@@ -5,15 +5,17 @@
...
 	<target host="maybank2u.com.ph" />
 	<target host="www.maybank2u.com.ph" />
+	
+	<securecookie host=".+" name=".+" />
 
-		<test url="http://maybank2u.com.ph/resources/my/img/common/logo-maybank_m.png" />
-		<test url="http://www.maybank2u.com.ph/resources/my/img/common/logo-maybank2u_m.png" />
-
-	<!-- 404 on main page -->
 	<exclusion pattern="^http://(www\.)?maybank2u\.com\.ph/$" />
```
[the remark this answers is not in the document]
Comment (youdly) [reply src/chrome/content/rules/Maybank2u.com.ph.xml]: Replaced with https redirect

File: src/chrome/content/rules/Maybank2u.com.ph.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+-->
+<ruleset name="Maybank2u.com.ph">
+
+	<target host="maybank2u.com.ph" />
+	<target host="www.maybank2u.com.ph" />
+	
+	<securecookie host=".+" name=".+" />
+
+	<!-- only main website; RIBNOW, resources and ph_maybank_gif directories are functional -->
+	<rule from="^http://(www\.)?maybank2u\.com\.ph/$"
```
Comment (J0WI) [inline src/chrome/content/rules/Maybank2u.com.ph.xml]: I get a 404 with this rule enabled.

Comment (youdly) [reply src/chrome/content/rules/Maybank2u.com.ph.xml]: WFM and travis

Comment (J0WI) [reply src/chrome/content/rules/Maybank2u.com.ph.xml]: @Bisaloo or @gloomy-ghost can you confirm this?

Comment (Bisaloo) [reply src/chrome/content/rules/Maybank2u.com.ph.xml]: I get 404 as well.

**Comment by J0WI** [comment]
@youdly this requires additional changes

**Comment by youdly** [comment]
Fixed

**Comment by J0WI** [comment]
Some tests are still failing

**Comment by youdly** [comment]
Fixed ruleset test coverage. Squashed commits.

**Pull request merged**